tornado-api/src/main/java/uk/ac/manchester/tornado/api/TornadoExecutionPlan.java

```diff
@@ -343,6 +347,7 @@ public TornadoExecutionPlan freeDeviceMemory() {
      */
     public TornadoExecutionPlan withGridScheduler(GridScheduler gridScheduler) {
         tornadoExecutor.withGridScheduler(gridScheduler);
+        executionFrame.withGridScheduler(gridScheduler);
```

Comment:
it is a bit strange that we set the tornadoExecutor both in line 349, and also in line 201. Are those two places mutually exclusive?